### PR TITLE
Manually update cuda migration

### PR DESCRIPTION
**Pull request by raulcd**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

I am trying to manually update the cuda migration and update the 14.0.2 URL to download Arrow to see if this fixes the build and the conda rerender.

### CODE REVIEW

**Comment by raulcd** [comment]
@conda-forge-admin, please rerender

File: .ci_support/migrations/cuda120.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ __migrator:
   kind:
     version
   migration_number:
-    1
+    3
```
Comment (h-vetinari) [inline .ci_support/migrations/cuda120.yaml:6]: No change to this file does anything as long as you don't specify `use_local: true` (which you shouldn't). Based on the name of presence of a given migrator file (by file name), conda smithy will always prefer the most current version of that migration in the global pinning.

**Comment by raulcd** [comment]
> Please don't mix two different things here - if you want to ensure 14.0.2 is rebuilt correctly, then let's do that. If you want to update/change something about the CUDA setup, then let's do that (but separately).

I'm just validating that the same issue that is happening on the 15.0.0 update is happening on main with the 14.0.2 branch. At the moment I'm just trying to understand what is the issue that has made the previous valid feedstock to start failing now.

File: .ci_support/migrations/cuda120.yaml (hunk 1 above)
Comment (raulcd) [reply .ci_support/migrations/cuda120.yaml:6]: If this file doesn't get updated comida smithy rerender was failing.

**Comment by h-vetinari** [comment]
> At the moment I'm just trying to understand what is the issue that has made the previous valid feedstock to start failing now.

The [logs](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=862392&view=results) of the CI after the last merge are very clear on that:
```
WARNING: Error: HTTP 404 NOT FOUND for url <[https://dist.apache.org/repos/dist/release/arrow/arrow-14.0.2/apache-arrow-14.0.2.tar.gz>](https://dist.apache.org/repos/dist/release/arrow/arrow-14.0.2/apache-arrow-14.0.2.tar.gz%3E)
```

File: .ci_support/migrations/cuda120.yaml (hunk 1 above)
Comment (h-vetinari) [reply .ci_support/migrations/cuda120.yaml:6]: Hm, strange. Can you try just updating the `migration_number` and nothing else?

**Comment by raulcd** [comment]
> The [logs](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=862392&view=results) of the CI after the last merge are very clear on that:

I think there is a small misunderstanding. I know the URL is missing but there is something else that is making the build fail. I have created a PR only updating the download URL to showcase: https://github.com/conda-forge/arrow-cpp-feedstock/pull/1292

**Comment by raulcd** [comment]
Closing this as not required.

**Pull request closed without merging**